src/graphql/graphql-api.php

```diff
@@ -0,0 +1,231 @@
+<?php
+/**
+ * GraphQL API
+ *
+ * @package vip-block-data-api
+ */
+
+namespace WPCOMVIP\BlockDataApi;
+
+defined( 'ABSPATH' ) || die();
+
+/**
+ * GraphQL API to offer an alternative to the REST API.
+ */
+class GraphQLApi {
+	/**
+	 * Initiatilize the graphQL API, if its allowed
+	 *
+	 * @access private
+	 */
+	public static function init() {
+		$is_graphql_to_be_enabled = apply_filters( 'vip_block_data_api__is_graphql_enabled', true );
+
+		if ( ! $is_graphql_to_be_enabled ) {
+			return;
+		}
+
+		add_filter( 'vip_block_data_api__sourced_block_result_transform', [ __CLASS__, 'transform_block_attributes' ], 10, 5 );
+
+		add_action( 'graphql_register_types', [ __CLASS__, 'register_types' ] );
+	}
+
+	/**
+	 * Extract the blocks data for a post, and return back in the format expected by the graphQL API.
+	 *
+	 * @param  WPGraphQL\Model\Post $post_model Post model for post.
+	 * @return array
+	 */
+	public static function get_blocks_data( $post_model ) {
+		$post_id        = $post_model->ID;
+		$post           = get_post( $post_id );
+		$filter_options = [ 'graphQL' => true ];
+
+		$content_parser = new ContentParser();
+
+		$parser_results = $content_parser->parse( $post->post_content, $post_id, $filter_options );
+
+		// ToDo: Verify if this is better, or is returning it another way in graphQL is better.
+		if ( is_wp_error( $parser_results ) ) {
+			// Return API-safe error with extra data (e.g. stack trace) removed.
+			return new \Exception( $parser_results->get_error_message() );
```

Comment:
Throwing WP_Error caused problems in WP-GraphQL, which is why I've done this.

Reply:
What kind of problem?

Reply:
![Screenshot 2023-12-08 at 17 20 31](https://github.com/Automattic/vip-block-data-api/assets/1771524/7cf10b7b-4058-4840-8c31-facdb358a590)

So if I return $parser_results as is, the above error is what I get. Maybe it should not be returned this way and I'm going about it incorrectly?

Reply:
![Screenshot 2023-12-08 at 17 22 30](https://github.com/Automattic/vip-block-data-api/assets/1771524/98979412-21b1-491e-8d6b-af77a13a507f)

vs the way I did it right now

Reply:
Code is fine as-is, was just curious